### PR TITLE
Sync diff from MS Docs and fix some bugs

### PR DESCRIPTION
**Pull request by hui1110**
1. Sync the changes from the following PRs:
   -  [PR about credential-free support: feature support ](https://github.com/MicrosoftDocs/azure-dev-docs-pr/pull/3329)
   -  [PR about credential-free support: quick start]( https://github.com/MicrosoftDocs/azure-dev-docs-pr/pull/3192)
   -  [PR about credential-free support: reference doc](https://github.com/MicrosoftDocs/azure-dev-docs-pr/pull/3242)

2. Fix the following issues:
   - In [Spring Cloud Azure Configuration](https://microsoft.github.io/spring-cloud-azure/current/reference/html/index.html#configuration), the link in `Table 8. Service specific properties` cannot jump to the properties table of the corresponding product.
   - Fix the problem of pipeline failure due to the `429` response. https://github.com/microsoft/spring-cloud-azure/runs/7229445253?check_suite_focus=true

### CODE REVIEW

**Comment by hui1110** [comment]
Hi @yiliuTo. @saragluna, please help review this PR. Thanks!

**Review by saragluna — DISMISSED** [review]
LGTM, @yiliuTo please take a look at this.

File: docs/src/main/asciidoc/kafka-support.adoc [hunk 1]
```diff
@@ -1,11 +1,150 @@
...
+ - Spring Kafka producer configuration options supersede the common options.
+ - Spring Kafka admin configuration options supersede the common options.
+ - The Spring Cloud Stream Kafka Binder options are just like the above.
+====
+
+==== Configurable properties when using Kafka support with plain connection string authentication
+
+
+.Spring Boot Kafka common configuration options
+[cols="2*", options="header"]
```
Comment (yiliuTo) [inline docs/src/main/asciidoc/kafka-support.adoc:134]: The table header should be `Spring Boot Event Hubs for Kafka common configuration options`

File: docs/src/main/asciidoc/kafka-support.adoc [hunk 2]
```diff
@@ -15,62 +154,138 @@ Add the following dependencies if you want to migrate your Apache Kafka applicat
...
 
-==== Use Azure Resource Manager to Retrieve Connection String
+If you want to retrieve the connection string using Azure Resource Manager, add the following dependency:
+
+[source,xml]
+----
+<dependency>
+  <groupId>com.azure.spring</groupId>
+  <artifactId>spring-cloud-azure-resourcemanager</artifactId>
+  </dependency>
```
Comment (yiliuTo) [inline docs/src/main/asciidoc/kafka-support.adoc:268]: the indentent here is not right, coudl you help to update the MS doc as well?

File: docs/src/main/asciidoc/kafka-support.adoc [hunk 3]
```diff
@@ -15,62 +154,138 @@ Add the following dependencies if you want to migrate your Apache Kafka applicat
...
+NOTE: Set `com.azure.spring.cloud.autoconfigure.kafka.AzureKafkaSpringCloudStreamConfiguration` to enable the whole OAuth authentication workflow.
 
-=== Basic Usage
+====== Samples
 
+See the link:https://github.com/Azure-Samples/azure-spring-boot-samples/tree/spring-cloud-azure_{project-version}[azure-spring-boot-samples] repository on GitHub.
 
-==== Use Event Hubs Connection String
+[#basic-usage-connection-string]
+==== Using connection string authentication
```
Comment (yiliuTo) [inline docs/src/main/asciidoc/kafka-support.adoc:243]: let's align all the grammer in this doc whether to use `use` or `using`

**Review by yiliuTo — CHANGES_REQUESTED** [review]

**Review by yiliuTo — APPROVED** [review]
LGTM

**Pull request merged**